tests/key_field/key_field_sample_generated.h

```diff
@@ -139,11 +139,11 @@ FLATBUFFERS_MANUALLY_ALIGNED_STRUCT(4) Bar FLATBUFFERS_FINAL_CLASS {
   bool KeyCompareLessThan(const Bar * const o) const {
     return KeyCompareWithValue(o->a()) < 0;
   }
-  int KeyCompareWithValue(const flatbuffers::Array<float, 3> *_a) const { 
+  int KeyCompareWithValue(const flatbuffers::Array<float, 3> *_a) const {
     for (auto i = 0; i < a()->size(); i++) {
-      const auto a_l = a_[i];
-      const auto a_r = _a->Get(i);
-      if(a_l != a_r) 
+      const auto a_l = flatbuffers::EndianScalar(a_[i]);
+      const auto a_r = flatbuffers::EndianScalar(_a->Get(i));
```

Comment:
If I am not mistaken,  the problem is exactly here, in KeyCompareWithValue method.
When stable_sort is called in CreateVectorOfSortedStructs on float vectors, KeyCompareWithValue  does not work properly in bigendian. Actually  flatbuffers::EndianScalar should just call on a_[i].

Furthermore, these changes are in the generated file, so when the project rebuild, it will be discard. These changes should be applied in cpp-generator.

Reply:
And, IMO if flatbuffers::Array<float, 3> could be used instead of float a_[3], It would be more safer.